### DISCUSSION TITLE
Link Neighbor Loader underlying logic

### DISCUSSION THREAD

**Discussion by phsla01** [pyg-team/pytorch_geometric#9768] [Q&A]
Hi, everyone. I'm using the MovieLensDataset for link prediction for undirected graph. I'm showing my data as `HeteroData` object and i'm using the `RandomLinkSplit` to create the train/val/test sets. Next, i use the `LinkNeighborLoader` to create batches and i'm showing only the test loader as an example here. From my understanding in the `edge_label_index` parameter i have to pass the message passing edges and in the `edge_label` a tensor with ones of the same size, so i'm not sure how the supervision edges are used here. The problem i encouter is when i examine the message passing edges (`mp_test_loader`) with the supervision edges (`sp_test_loader`) i have some common edges (target leakage), this only happens in the `test_loader` but not in the initial `test_data`. Any help why this is happening?

```python
HeteroData(
  user={ node_id=[610] },
  movie={
    node_id=[9742],
    x=[9742, 20],
  },
  (user, rates, movie)={ edge_index=[2, 100836] },
  (movie, rev_rates, user)={ edge_index=[2, 100836] }
)

transform = T.RandomLinkSplit(
    num_val=0.1,
    num_test=0.1,
    neg_sampling_ratio=2.0, 
    is_undirected=True,
    add_negative_train_samples=False,
    edge_types=("user", "rates", "movie"),
    rev_edge_types=("movie", "rev_rates", "user"),
)
train_data, val_data, test_data = transform(data)
 
edge_index = test_data["user", "rates", "movie"].edge_index
edge_label_index = test_data["user", "rates", "movie"].edge_label_index
edge_label = test_data["user", "rates", "movie"].edge_label
test_loader = LinkNeighborLoader(
    data=test_data,
    num_neighbors=[20, 10],
    edge_label_index=(("user", "rates", "movie"), edge_index),
    edge_label=torch.ones(edge_index.size(1)),
    batch_size= 128,
    shuffle=False,
    subgraph_type='induced'
)

mp_test_loader = next(iter(test_loader))['user', 'movie'].edge_index # message passing edges
sp_test_loader =  next(iter(test_loader))['user', 'movie'].edge_label_index # supervision edges (both positive/negative)
